### PR TITLE
ETag and nextLink middleware

### PR DESCRIPTION
**Pull request by mamaso**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@mamaso__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (mamaso). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by danderson00** [comment]
It has been verified that this conforms to V1 behavior?

**Comment by mamaso** [comment]
Mostly putting this up here for organizational purposes, I need to bug Phil & others about parity requirements

File: src/express/middleware/eTag.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// ----------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ----------------------------------------------------------------------------
+var queries = require('../../query');
+
+module.exports = {
+    singleResult: function (req, res, next) {
+        if (res.results && res.results.version) {
+            res.set('ETag', '"' + res.results.version + '"');
```
Comment (phvannor) [inline src/express/middleware/eTag.js]: is this already escaped for "

Comment (danderson00) [inline src/express/middleware/eTag.js]: I meant to mention that - this needs to do the opposite of utilities/strings/extractEtag

**Pull request closed without merging**